ScintSuite/SimulationCodes/SINPA/_Forw_Mod_xarray.py

```diff
@@ -0,0 +1,1713 @@
+                   maxval = True, figtitle = None, smap_val = None):
+    """
+    Plot one frame, the scintillator and the strikemap. 
+    
+    Alex Reyner: alereyvinn@alum.us.es
+
+    :param  frame dictionary containing:
+            'smap': smap used calibrated to the signal
+            'smapplt': smap extra to plot nice figures calibrated to the signal
+            'scintillator': scintillator calibrated to the signal
+            'signal_frame': signal in the scintillator space
+            'scint_area': region covered by the scintillator
+            'noises': all the different noises as matrices
+    :param  maxval: want to limit the colorbar to camera range?
+    :param  figtitle: plot a title with the optics parameters, for example
+
+    :return fig, ax:
+    """
+    print(" ----- 2 COLLIMATOR SIGNAL PLOT ----- ")
+    
+    frame_to_plot = frame['signal_frame']
+    smapplt = frame['smapplt']
+    scint_perim = frame['scintillator']
+
+    # Establish vmax as: 1) max counts, 2) 50% of the signal
+    if maxval == True:
+        max_count = 2 ** cam_params['range'] - 1
+        print('- Maximum set to camera range')
+    else:
+        max_count = frame_to_plot.max().item()*0.5
+        print('- Maximum set to 0.5 of the signal')
+
+    # Initialize the plot
+    fig, ax = plt.subplots(figsize=(7,4))
+    ax.set_aspect(1)      
+    
+    frame_to_plot.plot.imshow(ax=ax, cmap=ss.plt.Gamma_II(),
+                    vmin=0, vmax=max_count,
+                    cbar_kwargs={"label": 'Pixel counts','spacing': 'proportional'})
+
+    if plot_smap == True:
+        for item in smapplt:
+            smapplt[item].plot_pix(ax, labels=False, marker_params={'marker':None},
+                         line_params={'color':'w', 'linewidth':1.2, 'alpha':0.8})
+
+    if plot_scint == True:
+        ax.plot(scint_perim[:,0],scint_perim[:,1], color ='w', linewidth=3)
+
+    if plot_FoV == True:
+        try:
+            ax.scatter(frame['FoV_vect'][0], frame['FoV_vect'][1],
+                    marker='+',s=100,c='lime')
+            FoV = Circle((frame['FoV_vect'][0], frame['FoV_vect'][1]), 
+                         radius=frame['FoV_vect'][2], 
+                         color='lime', fill=False, linewidth=2)
+            ax.add_patch(FoV)
+        except:
+            print('- No FoV plotted beacuse whatever')
+
+    ax.set_xlim((0,cam_params['nx']))
+    ax.set_ylim((0,cam_params['ny']))
+    plt.xticks(fontsize=9)   
+    plt.yticks(fontsize=9)
+    ax_param = {'fontsize': 10, \
+                    'xlabel': 'x pix.', 'ylabel': 'y pix.'}
+    ax = ss.plt.axis_beauty(ax, ax_param)
+
+    if figtitle != None:
+        fig.suptitle(figtitle, fontsize=11)
+        
+    plt.show()
+
+    return fig, ax
+
+
+
+
+# -----------------------------------------------------------------------------
+# --- Auxiliary functions
+# -----------------------------------------------------------------------------
+
+def scint_ConvexHull(scint):
```

Comment:
Nice function, move it to the geometry file of the ScintSuite, so it is more visible
This is useful for all cases, not only for the xarray stuff